types/convert_test.go

```diff
@@ -50,329 +45,329 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) {
 	v, err = Convert("1,999,999.00", ft)
-	c.Assert(terror.ErrorEqual(err, ErrBadNumber), IsTrue, Commentf("err %v", err))
-	c.Assert(v.(*MyDecimal).String(), Equals, "1.0000")
+	require.Truef(t, terror.ErrorEqual(err, ErrBadNumber), "err %v", err)
+	require.Equal(t, "1.0000", v.(*MyDecimal).String())
 	v, err = Convert("199.00 ", ft)
-	c.Assert(err, IsNil)
-	c.Assert(v.(*MyDecimal).String(), Equals, "199.0000")
+	require.NoError(t, err)
+	require.Equal(t, "199.0000", v.(*MyDecimal).String())
 
 	// Test Datum.ToDecimal with bad number.
 	d := NewDatum("hello")
 	_, err = d.ToDecimal(sc)
-	c.Assert(terror.ErrorEqual(err, ErrBadNumber), IsTrue, Commentf("err %v", err))
+	require.Truef(t, terror.ErrorEqual(err, ErrBadNumber), "err %v", err)
 
 	sc.IgnoreTruncate = true
 	v, err = d.ToDecimal(sc)
-	c.Assert(err, IsNil)
-	c.Assert(v.(*MyDecimal).String(), Equals, "0")
+	require.NoError(t, err)
+	require.Equal(t, "0", v.(*MyDecimal).String())
 
 	// For TypeYear
 	ft = NewFieldType(mysql.TypeYear)
 	v, err = Convert("2015", ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, Equals, int64(2015))
+	require.NoError(t, err)
+	require.Equal(t, int64(2015), v)
 	v, err = Convert(2015, ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, Equals, int64(2015))
+	require.NoError(t, err)
+	require.Equal(t, int64(2015), v)
 	_, err = Convert(1800, ft)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 	dt, err := ParseDate(nil, "2015-11-11")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	v, err = Convert(dt, ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, Equals, int64(2015))
+	require.NoError(t, err)
+	require.Equal(t, int64(2015), v)
 	v, err = Convert(ZeroDuration, ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, Equals, int64(0))
+	require.NoError(t, err)
+	require.Equal(t, int64(0), v)
 	bj1, err := json.ParseBinaryFromString("99")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	v, err = Convert(bj1, ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, Equals, int64(1999))
+	require.NoError(t, err)
+	require.Equal(t, int64(1999), v)
 	bj2, err := json.ParseBinaryFromString("-1")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	_, err = Convert(bj2, ft)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 	bj3, err := json.ParseBinaryFromString("{\"key\": 99}")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	_, err = Convert(bj3, ft)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 	bj4, err := json.ParseBinaryFromString("[99, 0, 1]")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	_, err = Convert(bj4, ft)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 
 	// For enum
 	ft = NewFieldType(mysql.TypeEnum)
 	ft.Elems = []string{"a", "b", "c"}
 	v, err = Convert("a", ft)
-	c.Assert(err, IsNil)
-	c.Assert(v, DeepEquals, Enum{Name: "a", Value: 1})
+	require.NoError(t, err)
+	require.Equal(t, Enum{Name: "a", Value: 1}, v)
 	v, err = Convert(2, ft)
-	c.Log(errors.ErrorStack(err))
-	c.Assert(err, IsNil)
-	c.Assert(v, DeepEquals, Enum{Name: "b", Value: 2})
+	t.Log(errors.ErrorStack(err))
```

Comment:
It's a little puzzled why we need to log it?

Reply:
I'm not sure actually. Ideally `require.NoError(t, err)` will show `err` value if `err` is not `nil`. I saw that `c.Log(errors.ErrorStack(err))` simply logs the error if it's not `nil`, and prints empty string if it's `nil`. I just translated all the statements. But if you think it doesn't make sense for an extra log from `t.Log` apart from `require.NoError` in the case of `err` having a non `nil` value, then I can remove the `t.Log` line 👍

Reply:
Following is the `NoError` code:
```
func NoError(t TestingT, err error, msgAndArgs ...interface{}) bool {
	if err != nil {
		if h, ok := t.(tHelper); ok {
			h.Helper()
		}
		return Fail(t, fmt.Sprintf("Received unexpected error:\n%+v", err), msgAndArgs...)
	}

	return true
}
```
As you can see, `require.NoError(t, err)` will print an error log when err is not nil. So I think this `t.Log` is unnecessary.

Reply:
Cool, I can remove the extra log 👍

Reply:
And yes, I know that assertion frameworks print error log when assertion fails and I understood the `t.Log` was an extra log but since I wasn't the author of the test and just migrating it, I didn't wanna change anything without context

Reply:
Removed it 👍 Resolving this conversation